tools/OutlineService/OutlineService/OutlineService.cs

```diff
@@ -383,6 +406,29 @@ public partial class OutlineService : ServiceBase {
       }
     }
 
+    // Routes reserved and private subnets through the default gateway so they bypass the VPN.
+    private void AddReservedSubnetBypass(string systemGatewayIp, string interfaceName) {
+      try {
+        foreach (string subnet in IPV4_RESERVED_SUBNETS) {

```

Comment:
Let's think about exception handling. To date, `ConfigureRouting` has thrown an exception if **any** step fails - that's not the case here, as you're swallowing the exception.

Reply:
This was intentional - I don't think that it's worth failing a connection if we fail to exclude a particular subnet. I'm not sure that there is an equivalent step in `ConfigureRouting` at this point; if I'm not mistaken they are all necessary for the VPN to operate correctly, with setting the lower interface metric on the fence.

What would be the benefit of throwing and failing the connection?

Reply:
The benefit is that we don't leave the client in a weird state. I'm okay with that for this but let's update the comments for `ConfigureRouting`.